primitives/src/zeitgeist_currencies_extension.rs

```diff
@@ -0,0 +1,23 @@
+use crate::ZeitgeistMultiReservableCurrency;
+use frame_support::dispatch::DispatchResult;
+use orml_traits::{BasicCurrency, MultiCurrency};
+
+type CurrencyIdOf<T> = <<T as orml_currencies::Trait>::MultiCurrency as MultiCurrency<
+    <T as frame_system::Trait>::AccountId,
+>>::CurrencyId;
+
+pub trait ZeitgeistCurrenciesExtension: orml_currencies::Trait
+where
+    Self::MultiCurrency: ZeitgeistMultiReservableCurrency<Self::AccountId>,
+{
+    fn destroy_all(currency_id: CurrencyIdOf<Self>) -> DispatchResult {
+        let accounts = Self::MultiCurrency::accounts_by_currency_id(currency_id);
+        // Destroy pool
+        Self::MultiCurrency::destroy_all(currency_id, accounts.iter().cloned());
+        // Give back to accounts
+        for (k0, v) in accounts {
+            Self::NativeCurrency::deposit(&k0, v.free + v.reserved)?;
+        }
```

Comment:
```suggestion
```

This won't be needed. The operation that this is used for is when a market has been resolved and a whole `currency_id` should just simply be wiped from storage because they are the losing shares and therefore worthless.